vulkan/vulkan_swapchain.cc

```diff
@@ -0,0 +1,511 @@
+
+  if (!backbuffer->IsValid()) {
+    return nullptr;
+  }
+
+  current_backbuffer_index_ = next_backbuffer_index;
+  return backbuffer.get();
+}
+
+VulkanSwapchain::AcquireResult VulkanSwapchain::AcquireSurface() {
+  AcquireResult error = {AcquireType::ErrorSurfaceLost, nullptr};
+
+  if (!IsValid()) {
+    return error;
+  }
+
+  // ---------------------------------------------------------------------------
+  // Step 0:
+  // Acquire the next available backbuffer.
+  // ---------------------------------------------------------------------------
+  auto backbuffer = NextBackbuffer();
+
+  if (backbuffer == nullptr) {
+    return error;
+  }
+
+  // ---------------------------------------------------------------------------
+  // Step 1:
+  // Wait for use readiness.
+  // ---------------------------------------------------------------------------
+  if (!backbuffer->WaitFences()) {
+    return error;
+  }
+
+  // ---------------------------------------------------------------------------
+  // Step 2:
+  // Put semaphores in unsignled state.
+  // ---------------------------------------------------------------------------
+  if (!backbuffer->ResetFences()) {
+    return error;
+  }
+
+  // ---------------------------------------------------------------------------
+  // Step 3:
+  // Acquire the next image index.
+  // ---------------------------------------------------------------------------
+  uint32_t next_image_index = 0;
+
+  switch (VK_CALL_LOG_ERROR(
+      vk.AcquireNextImageKHR(device_.Handle(),                      //
+                             swapchain_,                            //
+                             std::numeric_limits<uint64_t>::max(),  //
+                             backbuffer->UsageSemaphore(),          //
+                             VK_NULL_HANDLE,                        //
+                             &next_image_index))                    //
+          ) {
+    case VK_SUCCESS:
+      break;
+    case VK_ERROR_OUT_OF_DATE_KHR:
+      return {AcquireType::ErrorSurfaceOutOfDate, nullptr};
+    default:
+      // Includes VK_ERROR_SURFACE_LOST_KHR.
+      return {AcquireType::ErrorSurfaceLost, nullptr};
+  }
+
+  // Simple sanity checking of image index.
+  if (next_image_index >= images_.size()) {
+    return error;
+  }
+
+  auto& image = images_[next_image_index];
+  if (!image->IsValid()) {
+    return error;
+  }
+
+  // ---------------------------------------------------------------------------
+  // Step 4:
+  // Start recording to the command buffer.
+  // ---------------------------------------------------------------------------
+  if (!backbuffer->UsageCommandBuffer().Begin()) {
+    return error;
+  }
+
+  // ---------------------------------------------------------------------------
+  // Step 5:
+  // Set image layout to color attachment mode.
+  // ---------------------------------------------------------------------------
+  VkPipelineStageFlagBits destination_pipline_stage =
```

Comment:
"pipeline", not "pipline" (Teams Canada and USA agree on this one).

Also, perhaps kDestinationPipelineStage and kDestinationImageLayout?